portfolio/portfolio_test.go

```diff
@@ -501,147 +278,129 @@ func TestGetPortfolioSummary(t *testing.T) {
 }
 
 func TestIsColdStorage(t *testing.T) {
 	t.Parallel()
-	newBase := seedPortFolioForTest(t)
-	ret := newBase.IsColdStorage(core.BitcoinDonationAddress)
-	if !ret {
-		t.Fatal("expected IsColdStorage() to return true")
-	}
-	ret = newBase.IsColdStorage(testBTCAddress)
-	if ret {
-		t.Fatal("expected IsColdStorage() to return false")
-	}
-	ret = newBase.IsColdStorage("hello")
-	if ret {
-		t.Fatal("expected IsColdStorage() to return false")
-	}
+	b := Base{
+		Addresses: []Address{
+			{
+				Address:     core.BitcoinDonationAddress,
+				ColdStorage: true,
+			},
+			{
+				Address: testBTCAddress,
+			},
+		},
+	}
+	assert.True(t, b.IsColdStorage(core.BitcoinDonationAddress), "IsColdStorage should return true for cold storage address")
+	assert.False(t, b.IsColdStorage(testBTCAddress), "IsColdStorage should return false for non-cold storage address")
 }
 
 func TestIsWhiteListed(t *testing.T) {
 	t.Parallel()
-	b := seedPortFolioForTest(t)
-	ret := b.IsWhiteListed(core.BitcoinDonationAddress)
-	if !ret {
-		t.Fatal("expected IsWhiteListed() to return true")
-	}
-	ret = b.IsWhiteListed(testBTCAddress)
-	if ret {
-		t.Fatal("expected IsWhiteListed() to return false")
-	}
-	ret = b.IsWhiteListed("hello")
-	if ret {
-		t.Fatal("expected IsWhiteListed() to return false")
-	}
+	b := Base{
+		Addresses: []Address{
+			{
+				Address:     core.BitcoinDonationAddress,
+				WhiteListed: true,
+			},
+			{
+				Address: testBTCAddress,
+			},
+		},
+	}
+	assert.True(t, b.IsWhiteListed(core.BitcoinDonationAddress), "IsWhiteListed should return true for whitelisted address")
+	assert.False(t, b.IsWhiteListed(testBTCAddress), "IsWhiteListed should return false for non-whitelisted address")
 }
 
 func TestStartPortfolioWatcher(t *testing.T) {
 	t.Parallel()
-	newBase := Base{}
-	err := newBase.AddAddress("LX2LMYXtuv5tiYEMztSSoEZcafFPYJFRK1",
-		currency.LTC.String(),
-		currency.NewCode(PersonalAddress),
-		0.02)
-	if err != nil {
-		t.Error(err)
-	}
+	b := Base{}
+	assert.ErrorIs(t, b.StartPortfolioWatcher(t.Context(), time.Second), errNoPortfolioItemsToWatch)
 
-	err = newBase.AddAddress("Testy",
-		currency.LTC.String(),
-		currency.NewCode(PersonalAddress),
-		0.02)
-	if err != nil {
-		t.Error(err)
-	}
+	assert.NoError(t, b.AddAddress(testXRPAddress, PersonalAddress, currency.XRP, 0.02))
 
-	if !newBase.AddressExists("LX2LMYXtuv5tiYEMztSSoEZcafFPYJFRK1") {
-		t.Error("address does not exist")
+	ctx, cancel := context.WithTimeout(t.Context(), time.Second)
+	defer cancel()
+
+	doneCh := make(chan error)
+	go func() {
+		doneCh <- b.StartPortfolioWatcher(ctx, 0) // 0 will set the ticker to 10 minutes
+	}()
+
+	select {
+	case err := <-doneCh:
+		assert.ErrorIs(t, err, context.DeadlineExceeded, "StartPortfolioWatcher should return DeadlineExceeded error")
+	case <-time.After(2 * time.Second):
+		assert.Fail(t, "StartPortfolioWatcher should have deadlined")
 	}
```

Comment:
```suggestion
	ctx, cancel := context.WithTimeout(t.Context(), 0)
	cancel()
	assert.ErrorIs(t, b.StartPortfolioWatcher(ctx, 0), context.DeadlineExceeded, "StartPortfolioWatcher should return DeadlineExceeded error")
```

Can cancel it straight away